dpnp/dpnp_iface_mathematical.py

```diff
@@ -1682,51 +1845,206 @@ def fmod(x1, x2, /, out=None, *, where=True, dtype=None, subok=True, **kwargs):
     )
 
 
-def gradient(x1, *varargs, **kwargs):
+def gradient(f, *varargs, axis=None, edge_order=1):
     """
-    Return the gradient of an array.
+    Return the gradient of an N-dimensional array.
+
+    The gradient is computed using second order accurate central differences
+    in the interior points and either first or second order accurate one-sides
+    (forward or backwards) differences at the boundaries.
+    The returned gradient hence has the same shape as the input array.
 
     For full documentation refer to :obj:`numpy.gradient`.
 
-    Limitations
-    -----------
-    Parameter `y1` is supported as :class:`dpnp.ndarray`.
-    Argument `varargs[0]` is supported as `int`.
-    Keyword argument `kwargs` is currently unsupported.
-    Otherwise the function will be executed sequentially on CPU.
-    Input array data types are limited by supported DPNP :ref:`Data types`.
+    Parameters
+    ----------
+    f : {dpnp.ndarray, usm_ndarray}
+        An N-dimensional array containing samples of a scalar function.
+    varargs : {scalar, list of scalars, list of arrays}, optional
```

Comment:
list of scalars and list of arrays doesn`t support
```
In [1]: import dpnp

In [2]: f = dpnp.array([1, 2, 4, 7, 11, 16], dtype='float32')

In [3]: x = dpnp.arange(f.size)

In [4]: dpnp.gradient(f,x)
Out[4]: array([1. , 1.5, 2.5, 3.5, 4.5, 5. ], dtype=float32)

In [5]: dpnp.gradient(f,[0,1,2,3,4,5])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[5], line 1
----> 1 dpnp.gradient(f,[0,1,2,3,4,5])

File ~/work/orig/dpnp/dpnp/dpnp_iface_mathematical.py:1974, in gradient(f, axis, edge_order, *varargs)
   1971 else:
   1972     axes = normalize_axis_tuple(axis, ndim)
-> 1974 dx = _gradient_build_dx(f, axes, *varargs)
   1975 if edge_order > 2:
   1976     raise ValueError("'edge_order' greater than 2 not supported")

File ~/work/orig/dpnp/dpnp/dpnp_iface_mathematical.py:190, in _gradient_build_dx(f, axes, *varargs)
    188 dx = list(varargs)
    189 for i, distances in enumerate(dx):
--> 190     dpnp.check_supported_arrays_type(
    191         distances, scalar_type=True, all_scalars=True
    192     )
    194     if numpy.ndim(distances) == 0:
    195         continue

File ~/work/orig/dpnp/dpnp/dpnp_iface.py:330, in check_supported_arrays_type(scalar_type, all_scalars, *arrays)
    327     if scalar_type and dpnp.isscalar(a):
    328         continue
--> 330     raise TypeError(
    331         f"An array must be any of supported type, but got {type(a)}"
    332     )
    334 if len(arrays) > 0 and not (all_scalars or any_is_array):
    335     raise TypeError(
    336         "At least one input must be of supported array type, "
    337         "but got all scalars."
    338     )

TypeError: An array must be any of supported type, but got <class 'list'>
```

Reply:
you need to unpack the list to have that working, like:
```python
dpnp.gradient(f, 0, 1, 2, 3, 4, 5)
```
since the signature is `def gradient(f, *varargs, axis=None, edge_order=1)` where `varargs` is defined with asterisk.